pinot-spi/src/main/java/org/apache/pinot/spi/plugin/PluginClassLoader.java

```diff
@@ -36,18 +35,16 @@
   public PluginClassLoader(URL[] urls, ClassLoader parent) {
     super(urls, parent);
     classLoader = PluginClassLoader.class.getClassLoader();
-    Method method = null;
-    try {
-      method = URLClassLoader.class.getDeclaredMethod("addURL", URL.class);
-
-    } catch (NoSuchMethodException e) {
-      //this should never happen
-      ExceptionUtils.rethrow(e);
-    }
-    method.setAccessible(true);
     for (URL url : urls) {
       try {
-        method.invoke(classLoader, url);
+        /**
+         * ClassLoader in java9+ does not extend URLClassLoader.
```

Comment:
Don't quite understand the comment here. Do you mean `URLClassLoader` does not extend `ClassLoader` in java 9? But it should still have the method `addURL()` right?

Reply:
`URLClassLoader` still extend `ClassLoader` and this `addURL` method is there.
Actually, this part also works for java 8. 
@kishoreg @elonazoulay thoughts?